libraries/CurieBLE/src/internal/BLECharacteristicImp.h

```diff
@@ -324,7 +324,7 @@ class BLECharacteristicImp: public BLEAttribute{
     bt_gatt_subscribe_params_t _sub_params;
     bool        _subscribed;
     
-    bool _reading;
+    volatile bool _reading;
```

Comment:
Please note last two changes in BLECharacteristicImp.cpp. It need to make compiler not optimumize the code.